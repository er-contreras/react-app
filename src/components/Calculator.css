#calc-container {
  display: grid;
  grid-template-columns: repeat(4, 120px);
  grid-template-rows: minmax(70px, auto) repeat(5, 70px);
  background-color: rgb(166, 166, 166);
}

#output {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  grid-column: span 4;
  font-size: 30px;
  background-color: rgb(133, 134, 148);
  color: white;
  padding: 5px;
}

#zero {
  grid-column: span 2;
}

#calc-container > button {
  cursor: pointer;
  font-size: 1.5rem;
  border: 1px solid rgb(201, 201, 201);
}

.equals,
.add-operator,
.subtract-operator,
.multiply-operator,
.divide-operator {
  background-color: rgb(255, 140, 0);
  color: rgb(0, 0, 0);
  font-size: 2rem;
}
