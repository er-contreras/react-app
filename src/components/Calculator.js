import React, { useState } from 'react';
import './Calculator.css';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [initialState, updateState] = useState({ total: null, next: null, operation: null });

  const handleClick = (buttonName) => {
    const { total, next, operation } = calculate(initialState, buttonName);
    updateState({ total, next, operation });
  };

  const updateDisplay = () => {
    let { total, next, operation } = initialState;
    total = total || '';
    operation = operation || '';
    next = next || '';
    return `${total} ${operation} ${next}`;
  };

  return (
    <>
      <div className="lets-do">
        <p>
          Lets do some math!
        </p>
      </div>

      <div id="calc-container">
        <div id="output">
          {updateDisplay()}
        </div>
        <button onClick={() => handleClick('AC')} type="button" className="ac">AC</button>
        <button onClick={() => handleClick('+/-')} type="button" className="quantity">+/-</button>
        <button onClick={() => handleClick('%')} type="button" className="percentage-operator">%</button>
        <button onClick={() => handleClick('รท')} type="button" className="divide-operator">รท</button>
        <button onClick={() => handleClick('7')} type="button" className="number">7</button>
        <button onClick={() => handleClick('8')} type="button" className="number">8</button>
        <button onClick={() => handleClick('9')} type="button" className="number">9</button>
        <button onClick={() => handleClick('x')} type="button" className="multiply-operator">x</button>
        <button onClick={() => handleClick('4')} type="button" className="number">4</button>
        <button onClick={() => handleClick('5')} type="button" className="number">5</button>
        <button onClick={() => handleClick('6')} type="button" className="number">6</button>
        <button onClick={() => handleClick('-')} type="button" className="subtract-operator">-</button>
        <button onClick={() => handleClick('1')} type="button" className="number">1</button>
        <button onClick={() => handleClick('2')} type="button" className="number">2</button>
        <button onClick={() => handleClick('3')} type="button" className="number">3</button>
        <button onClick={() => handleClick('+')} type="button" className="add-operator">+</button>
        <button onClick={() => handleClick('0')} type="button" id="zero" className="number">0</button>
        <button onClick={() => handleClick('.')} type="button" className="number">.</button>
        <button onClick={() => handleClick('=')} type="button" className="equals">=</button>
      </div>
    </>
  );
};

export default Calculator;
