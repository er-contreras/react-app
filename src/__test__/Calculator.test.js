import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('Calculator test', () => {
  it('Should render same text passed into ...', () => {
    const tree = TestRenderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the main heading', () => {
    render(<Calculator />);
    const heading = screen.getByText(/Lets do some math/i);
    expect(heading).toMatchSnapshot();
  });

  it('Test AC', () => {
    render(<Calculator />);
    const allClear = screen.getByText(/AC/i);
    expect(allClear).toMatchSnapshot();
  });

  it('Test +/-', () => {
    render(<Calculator />);
    const simbol = screen.getByText('+/-');
    expect(simbol).toMatchSnapshot();
  });

  it('Test %', () => {
    render(<Calculator />);
    const quantity = screen.getByText(/%/i);
    expect(quantity).toMatchSnapshot();
  });
})