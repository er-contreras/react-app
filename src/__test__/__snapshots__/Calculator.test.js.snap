// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator test Should render same text passed into ... 1`] = `
Array [
  <div
    className="lets-do"
  >
    <p>
      Lets do some math!
    </p>
  </div>,
  <div
    id="calc-container"
  >
    <div
      id="output"
    >
        
    </div>
    <button
      className="ac"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="quantity"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="percentage-operator"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="divide-operator"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="multiply-operator"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="subtract-operator"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="add-operator"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="number"
      id="zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="number"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="equals"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>,
]
`;

exports[`Calculator test Test % 1`] = `
<button
  class="percentage-operator"
  type="button"
>
  %
</button>
`;

exports[`Calculator test Test +/- 1`] = `
<button
  class="quantity"
  type="button"
>
  +/-
</button>
`;

exports[`Calculator test Test AC 1`] = `
<button
  class="ac"
  type="button"
>
  AC
</button>
`;

exports[`Calculator test renders the main heading 1`] = `
<p>
  Lets do some math!
</p>
`;
